### ISSUE TITLE
Repo over 4Gb

### ISSUE THREAD

**Issue by levinuss** [mailcow/mailcow-dockerized#1018]
Hey,
I would like to move the mailcow-dockerized dir with the dockerfiles etc to my SSD so I looked at its size with `du -skh /path/to/mailcow`. I got something about 4.4 Gb, even though the actual files are only some Mb. The rest is only the git repo (most of it the .pack file in .git/objects/pack).
I tried some versions of `git fsck`, `git prune` and `git repack` after reading some stackoverflow questions about large git repos. These commands didn't help, I'm now at 4.3Gb.
Because nothing really worked I'm now asking here. What could the reason be? I used the manual update method for some time now, I recently changed my update mechanism to just run the update.sh. The installation on that machine was made in June 2017. Is this the normal behavior?

**Comment by andryyy** [comment]
Hi,
Have you tried `git gc --aggressive --prune` ?

**Comment by levinuss** [comment]
Yes I also tried `git gc`
I suspect also with these parameters. To be sure I just ran your command, some result. Repo remains 4.3Gb large.

**Comment by MAGICCC** [comment]
Can you check what files are the biggest ones with the following script? You may install `bsdultils` for the `column` command in the last echo:
```
#!/bin/bash
#set -x 

# Shows you the largest objects in your repo's pack file.
# Written for osx.
#
# @see http://stubbisms.wordpress.com/2009/07/10/git-script-to-show-largest-pack-objects-and-trim-your-waist-line/
# @author Antony Stubbs

# set the internal field spereator to line break, so that we can iterate easily over the verify-pack output
IFS=$'\n';

# list all objects including their size, sort by size, take top 10
objects=`git verify-pack -v .git/objects/pack/pack-*.idx | grep -v chain | sort -k3nr | head`

echo "All sizes are in kB. The pack column is the size of the object, compressed, inside the pack file."

output="size,pack,SHA,location"
for y in $objects
do
	# extract the size in bytes
	size=$((`echo $y | cut -f 5 -d ' '`/1024))
	# extract the compressed size in bytes
	compressedSize=$((`echo $y | cut -f 6 -d ' '`/1024))
	# extract the SHA
	sha=`echo $y | cut -f 1 -d ' '`
	# find the objects location in the repository tree
	other=`git rev-list --all --objects | grep $sha`
	#lineBreak=`echo -e "\n"`
	output="${output}\n${size},${compressedSize},${other}"
done

echo -e $output | column -t -s ', '
```

**Comment by levinuss** [comment]
@MAGICCC Thank you for the script! I didn't find that one! It took some time to compute but it showed me that some old backups I made into the mailcow dir were committed to the repo. I didn't have in mind that all files in the mailcow dir are staged...
I cloned the mailcow dir to a new directory and copied everything except the .git dir to the new repo. After running update.sh, mailcow uses the files from the new repo (it should?).
